tests/functional/api/groups/members_test.py

```diff
@@ -62,179 +62,135 @@ def test_it_returns_empty_list_if_no_members_in_group(self, app):
+    def test_it_errors_if_the_userid_is_invalid(self, do_request):
+        do_request(userid="INVALID_USERID", status=404)
 
-    def test_it_ignores_forwarded_user_header(
-        self, app, factories, db_session, auth_client
-    ):
-        third_party_user = factories.User(authority="thirdparty.com")
-        third_party_group = factories.Group(authority="thirdparty.com")
-        headers = self.authorization_header(auth_client)
-        user2 = factories.User(authority="thirdparty.com")
-        db_session.commit()
+    def test_it_errors_if_the_request_isnt_authenticated(self, do_request, headers):
+        del headers["Authorization"]
 
-        headers["X-Forwarded-User"] = third_party_user.userid
+        do_request(status=404)
 
-        res = app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=third_party_group.pubid, userid=third_party_user.userid
-            ),
-            headers=headers,
-        )
+    def test_it_errors_if_the_request_has_token_authentication(
+        self, factories, user, headers, do_request
+    ):
+        token = factories.DeveloperToken(user=user)
+        headers.update(token_authorization_header(token))
 
-        assert third_party_user in third_party_group.members
-        assert user2 not in third_party_group.members
-        assert res.status_code == 204
+        do_request(status=404)
 
-    def test_it_is_idempotent(self, app, auth_client, db_session, factories):
-        third_party_user = factories.User(authority="thirdparty.com")
-        third_party_group = factories.Group(authority="thirdparty.com")
-        db_session.commit()
+    def test_it_errors_if_the_groups_authority_doesnt_match(
+        self, factories, do_request
+    ):
+        group = factories.Group(authority="other")
 
-        app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=third_party_group.pubid, userid=third_party_user.userid
-            ),
-            headers=self.authorization_header(auth_client),
-        )
+        do_request(pubid=group.pubid, status=404)
 
-        res = app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=third_party_group.pubid, userid=third_party_user.userid
-            ),
-            headers=self.authorization_header(auth_client),
-        )
+    def test_it_errors_if_the_users_authority_doesnt_match(self, factories, do_request):
+        user = factories.User(authority="other")
 
-        assert third_party_user in third_party_group.members
-        assert res.status_code == 204
+        do_request(userid=user.userid, status=404)
 
-    def test_it_returns_404_if_authority_mismatch_on_user(
-        self, app, factories, auth_client
+    def test_it_errors_if_the_groups_and_users_authorities_both_dont_match(
+        self, factories, do_request
     ):
-        group = factories.Group()
-        user = factories.User(authority="somewhere-else.org")
```

Comment:
`group` has wrong authority and DB session isn't committed, so test is not testing what it intends to.